src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,244 @@
+package duke.parser;
+
+import duke.Deadline;
+import duke.Event;
+import duke.Todo;
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.InvalidCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.TodoCommand;
+import duke.command.UnmarkCommand;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static duke.common.Messages.MESSAGE_INVALID_COMMAND;
+
+/**
+ * Parses user input into commands.
+ */
+public class Parser {
+    private static final Pattern COMMAND_FORMAT = Pattern.compile("^(?<command>\\S+)(?<arguments>.*)$");
+
+    public static Todo parseTodo(String args) {
+        return new Todo(args);
+    }
+
+    /**
+     * Parses a deadline input string into a Deadline object.
+     *
+     * @param args the input string for the deadline
+     * @return the parsed Deadline object
+     * @throws DukeParseException if unable to parse correctly
+     */
+    public static Deadline parseDeadline(String args) throws DukeParseException {
+        Pattern pattern = Pattern.compile("^(?<desc>.+) /by (?<by>.+)$");
+        Matcher matcher = pattern.matcher(args);
+        if (!matcher.matches()) {
+            throw new DukeParseException("Deadline cannot be parsed");
+        }
+        String desc = matcher.group("desc").trim();
+        String by = matcher.group("by").trim();
+        try {
+            LocalDate byDate = LocalDate.parse(by);
+            return new Deadline(desc, byDate);
+        } catch (DateTimeParseException e) {
+            throw new DukeParseException("Deadline cannot be parsed");
+        }
+    }
+
+    /**
+     * Parses an event input string into an Event object.
+     *
+     * @param args the input string for the event
+     * @return the parsed Event object
+     * @throws DukeParseException if unable to parse correctly
+     */
+    public static Event parseEvent(String args) throws DukeParseException {
+        Pattern pattern = Pattern.compile("^(?<desc>.+) /from (?<from>.+) /to (?<to>.+)$");
+        Matcher matcher = pattern.matcher(args);
+        if (!matcher.matches()) {
+            throw new DukeParseException("Event cannot be parsed");
+        }
+        String desc = matcher.group("desc").trim();
+        String from = matcher.group("from").trim();
```

Comment:
It could be helpful for readability if the logical units within each code block are separated by one blank line, especially when methods are slightly more complex